File: Src/java/engine/src/main/java/org/opencds/cqf/cql/engine/elm/executing/ToListEvaluator.java
package org.opencds.cqf.cql.engine.elm.executing;

import java.util.Collections;

public class ToListEvaluator {

    public static Object toList(Object operand)
    {
        // check to see if it is already a list
        if (operand instanceof Iterable)
        {
            return operand;
        }

        return operand == null ? Collections.emptyList() : Collections.singletonList(operand);
    }
}

File: Src/java/engine/src/main/kotlin/org/opencds/cqf/cql/engine/elm/executing/ToListEvaluator.kt
package org.opencds.cqf.cql.engine.elm.executing

object ToListEvaluator {
    @JvmStatic
    fun toList(operand: Any?): Any? {
        // check to see if it is already a list
        if (operand is Iterable<*>) {
            return operand
        }

        return if (operand == null) mutableListOf<Any?>() else mutableListOf<Any?>(operand)
    }
}
